src/main/java/nl/tudelft/cse1110/andy/execution/security/AndySecurityManager.java

```diff
@@ -0,0 +1,182 @@
+package nl.tudelft.cse1110.andy.execution.security;
+
+import java.io.FilePermission;
+import java.lang.reflect.ReflectPermission;
+import java.net.NetPermission;
+import java.net.SocketPermission;
+import java.net.URLPermission;
+import java.security.Permission;
+import java.security.SecurityPermission;
+import java.util.PropertyPermission;
+import java.util.logging.LoggingPermission;
+
+public class AndySecurityManager extends SecurityManager {
+    public AndySecurityManager() {
+        super();
+    }
+
+    @Override
+    public void checkPermission(Permission perm) {
+        boolean untrusted = false;
+        boolean mockitoInternal = false;
+        boolean jdkInternalLoader = false;
+        for (StackTraceElement elem : Thread.currentThread().getStackTrace()) {
+            if (elem.getClassName().startsWith("org.mockito.internal")) {
+                mockitoInternal = true;
+            }
+            if (elem.getClassName().startsWith("jdk.internal.") && !untrusted) {
+                jdkInternalLoader = true;
+            }
+            if (elem.getClassName().startsWith("delft.")) {
+                untrusted = true;
+            }
+        }
+
+        if (!untrusted) {
+            return;
+        }
+
+        if (checkPermissionsUntrusted(perm, mockitoInternal, jdkInternalLoader)) return;
+
+        var ex = new SecurityException("Operation not permitted: " +
+                                       perm.getClass().getName() + " " +
+                                       perm.getName() + " " +
+                                       perm.getActions());
+        ex.printStackTrace();
```

Comment:
I put it here so that we can more easily debug permission issues. In some situations, Andy simply ignores the exception (especially if that exception occurs while running one of the tools). I left it here in case there are more platform-dependent issues that occur once this is deployed on weblab.

Reply:
Maybe we could add an environment variable to determine if this exception should be printed. Do we have any control over the environment variables passed to Andy on WebLab?

Reply:
No control. But if this is a rare event, we can leave this print here.